src/_ert/forward_model_runner/forward_model_step.py

```diff
@@ -427,6 +419,26 @@ def ensure_file_handles_closed(file_handles: Sequence[io.TextIOWrapper | None])
             file_handle.close()
 
 
+@dataclass
+class ProcesstreeTimer:
+    _cpu_seconds_pr_pid: dict[str, float] = field(default_factory=dict, init=False)
+
+    def update(self, cpu_seconds_snapshot: dict[str, float]) -> None:
+        for pid, seconds in cpu_seconds_snapshot.items():
+            if self._cpu_seconds_pr_pid.get(str(pid), 0.0) <= seconds:
+                self._cpu_seconds_pr_pid[str(pid)] = seconds
+            else:
+                # cpu_seconds must be monotonely increasing. Since
+                # decreasing cpu_seconds was detected, it must be due to pid reuse
+                self._cpu_seconds_pr_pid[str(pid) + str(uuid.uuid4())] = (
+                    self._cpu_seconds_pr_pid[str(pid)]
+                )
+                self._cpu_seconds_pr_pid[str(pid)] = seconds
```

Comment:
This can probably be even smaller.

```
if self._cpu_seconds_pr_pid.get(str(pid), 0.0) > seconds:
    # cpu_seconds must be monotonely increasing. Since
    # decreasing cpu_seconds was detected, it must be due to pid reuse
    self._cpu_seconds_pr_pid[str(pid) + str(uuid.uuid4())] = (
        self._cpu_seconds_pr_pid[str(pid)]
    )

self._cpu_seconds_pr_pid[str(pid)] = seconds

Reply:
Not sure how but there is a get and set here so seems like you might get away with a `setdefault` ?